OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -79,6 +80,17 @@ static class DreamProcNativeRoot {
 
             return new DreamValue(Convert.ToChar(ascii).ToString());
         }
+        
+        [DreamProc("clamp")]
+        [DreamProcParameter("Value", Type = DreamValueType.Number)]
+        [DreamProcParameter("Low", Type = DreamValueType.Number)]
+        [DreamProcParameter("High", Type = DreamValueType.Number)]
+        public static DreamValue NativeProc_ckey(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            DreamValue iValue = arguments.GetArgument(0, "Value");
+            DreamValue lowValue = arguments.GetArgument(1, "Low");
+            DreamValue highValue = arguments.GetArgument(2, "High");
+            return Math.Clamp(iValue, lowValue, highValue);
```

Comment:
Arguments are DreamValues, and have to be unwrapped. Similarly, a proc's return value must be wrapped in a DreamValue.

```cs
float iValue = arguments.GetArgument(0, "Value").GetValueAsNumber();
float lowValue = arguments.GetArgument(1, "Low").GetValueAsNumber();
float highValue = arguments.GetArgument(2, "High").GetValueAsNumber();
return new DreamValue(Math.Clamp(iValue, lowValue, highValue));
```